offlinevalidator/offlinevalidator.go

```diff
@@ -87,11 +85,51 @@ func Validate(context *context.Context) bool {
 	resources := obtainResources(deNilizedTemplate, perunTemplate, context.Logger)
 	deadResources := getNilResources(resources)
 	deadProperties := getNilProperties(resources)
+	hasAllowedValues := findingAllowedValues(goFormationTemplate, context.Logger)
+	if hasAllowedValues == true {
+		return false
+	}
 
 	valid = validateResources(resources, &specification, context.Logger, deadProperties, deadResources)
 	return valid
 }
 
+func findingAllowedValues(goFormationTemplate cloudformation.Template, logger *logger.Logger) bool {
+	parameters := goFormationTemplate.Parameters
+	isType := false
+	isAllovedValues := false
+
+	for _, value := range parameters {
+		valueof := reflect.ValueOf(value)
+		if valueof.Kind() == reflect.Map {
+			isAllovedValues = false
+			isType = false
+			for _, key := range valueof.MapKeys() {
+				strct := valueof.MapIndex(key)
+				textType := "Type"
+				keyString := key.Interface().(string)
+				textValues := "AllowedValues"
+
+				if textType == keyString {
+					textString := "String"
+					strctString := strct.Interface().(string)
+					if textString != strctString {
```

Comment:
Shouldn't it be `strict`?

Reply:
What do you mean? Do you want to this: `if textType == keyString && textString != strctString` ?

Reply:
Sorry, for not being clear here - `strct` is a short form from something or a typo from `strict`? :wink:

Reply:
It's `structure`. I'll rename it if it's confusing :)